### PR TITLE
Improve the default settings

### PR DESCRIPTION
**Pull request by EbenZhang**

### CODE REVIEW

File: GitCommands/Settings/AppSettings.cs [hunk 1]
```diff
@@ -615,7 +615,7 @@ public static bool DontConfirmAmend
 
         public static bool DontConfirmPushNewBranch
         {
-            get { return GetBool("DontConfirmPushNewBranch", false); }
+            get { return GetBool("DontConfirmPushNewBranch", true); }
```
Comment (jbialobr) [inline GitCommands/Settings/AppSettings.cs]: Should it be moved from the Advanced to the Detailed tab?

File: GitCommands/Settings/AppSettings.cs [hunk 2]
```diff
@@ -991,7 +991,7 @@ public static bool IgnoreWhitespaceChanges
 
         public static bool RememberIgnoreWhiteSpacePreference
         {
-            get { return GetBool("rememberIgnoreWhiteSpacePreference", false); }
+            get { return GetBool("rememberIgnoreWhiteSpacePreference", true); }
```
Comment (jbialobr) [inline GitCommands/Settings/AppSettings.cs]: Should it be moved from the Advanced to the Detailed tab?

File: GitCommands/Settings/AppSettings.cs (hunk 1 above)
Comment (EbenZhang) [inline GitCommands/Settings/AppSettings.cs]: > Should it be moved from the Advanced to the Detailed tab?

What is the difference between `Advanced` and `Detailed`?

Anyway it's nothing to do with my changes in this PR.

Comment (jbialobr) [inline GitCommands/Settings/AppSettings.cs]: The Advanced settings page should contain settings that require advanced knowledge to change the default value of these settings. Such a change is considered risky and should be done with caution.
By no means unchecking the DontConfirmPushNewBranch option can not be considered risky or requiring advanced knowledge.

**Pull request closed without merging**